django/common/adapters/fhir_api.py

```diff
@@ -0,0 +1,30 @@
+from django.conf import settings
+
+import requests
+
+
+class FhirAPI:
+    def __init__(self, auth_token=None):
+        self._auth_token = auth_token
+        self._headers = (
+            {"Cache-Control": "no-cache", "Authorization": f"Bearer {auth_token}"}
+            if auth_token
+            else {"Cache-Control": "no-cache"}
+        )
+
+    def get(self, path):
+        raise NotImplementedError
+
+
+class HapiFhirAPI(FhirAPI):
+    def __init__(self, auth_token=None):
+        super().__init__(auth_token)
+        self._url = settings.FHIR_API_URL
+
+    def get(self, path):
+        response = requests.get(
+            f"{self._url}{path}",
```

Comment:
maybe we could use path.join here ? it would avoid missing or double slash issues

Reply:
os.path.join ? the function I found for url is urljoin from urllib.parse and it doesn't work well in that case since it takes only the base path of self._url (http://something/ not http://something/hapi/fhir)